examples/wave-mpi.py

```diff
@@ -217,14 +220,15 @@ def rhs(t, w):
             print(istep, t, discr.norm(fields[0]))
             vis.write_parallel_vtk_file(
                 comm,
-                "fld-wave-eager-mpi-%03d-%04d.vtu" % (rank, istep),
+                "fld-wave-mpi-%03d-%04d.vtu" % (rank, istep),
                 [
                     ("u", fields[0]),
                     ("v", fields[1:]),
                 ]
             )
 
-        fields = rk4_step(fields, t, dt, rhs)
+        fields = thaw(freeze(fields, actx), actx)
+        fields = rk4_step(fields, t, dt, compiled_rhs)
```

Comment:
Should we add these type of changes in the other example drivers?

Reply:
We've also added these inside the `advance_timestepper` function. As long as the drivers use that helper, there shouldn't be any need to add it again.